### PR TITLE
feat(form): add support for anchor links

### LINKED ISSUE #23: Add HTML anchors to form items
Add HTML anchors to form items so that we can reference and auto-scroll directly to the linked item

### PR DESCRIPTION
**Pull request by miguelgrc**
Closes #23 

- Shows anchor icon on hover over field titles
- Anchor links work also with tab fields
- Can be disabled globally by passing `hideAnchors` to FormuleForm or to FormPreview

### CODE REVIEW

File: src/exposed.tsx [hunk 1]
```diff
@@ -49,6 +51,7 @@ export const FormuleContext = ({
               customFields,
               customWidgets,
               transformSchema,
+              hideAnchors,
```
Comment (pamfilos) [inline src/exposed.tsx:54]: why do we need it here?

File: src/forms/fields/internal/TitleField.jsx [hunk 2]
```diff
@@ -23,7 +26,10 @@ const TitleField = ({
   readonly,
   titleIsMarkdown,
   isObject,
+  hideAnchor,
```
Comment (pamfilos) [inline src/forms/fields/internal/TitleField.jsx:29]: might be better to pass hideAnchor for `formContext` straight from RJSF

File: src/forms/fields/internal/TitleField.jsx [hunk 3]
```diff
@@ -50,8 +57,8 @@ const TitleField = ({
   if (!title) return null;
 
   const titleText = (
-    <Typography.Text
-      style={{ fontSize: isObject && "12pt" }}
+    <Typography.Paragraph
```
Comment (pamfilos) [inline src/forms/fields/internal/TitleField.jsx:60]: why change?

File: src/forms/templates/Field/FieldTemplate.jsx [hunk 4]
```diff
@@ -29,6 +29,7 @@ const FieldTemplate = ({
   schema,
   uiSchema = {},
   rawDescription,
+  hideAnchor,
```
Comment (pamfilos) [inline src/forms/templates/Field/FieldTemplate.jsx:32]: again can be passed from `formContext`

**Review by pamfilos — CHANGES_REQUESTED** [review]

File: src/exposed.tsx (hunk 1 above)
Comment (miguelgrc) [reply src/exposed.tsx:54]: For users to be able to decide if they want to disable anchors

File: src/forms/fields/internal/TitleField.jsx (hunk 3 above)
Comment (miguelgrc) [reply src/forms/fields/internal/TitleField.jsx:60]: The html element generated by `Text` has the same width as the text, so you need to hover over it to reveal the anchor icon. With `Paragraph` it expands to fill all the row, so the icon would be revealed by hovering anywhere on the row, which I like more.

**Review by pamfilos — APPROVED** [review]

**Pull request merged**